Makefile

```diff
@@ -185,121 +199,398 @@ test-integration: test-integration-aws test-integration-clickhouse test-integrat
+start-test-integration: start-integration-gcp start-integration-influxdb start-integration-kafka start-integration-loki
+start-test-integration: start-integration-pulsar start-integration-splunk
+
+stop-test-integration: ## Stops all integration test infrastructure
+stop-test-integration: stop-integration-aws stop-integration-clickhouse stop-integration-elasticsearch
+stop-test-integration: stop-integration-gcp stop-integration-influxdb stop-integration-kafka stop-integration-loki
+stop-test-integration: stop-integration-pulsar stop-integration-splunk
+
+start-integration-aws:
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create --replace --name vector-test-integration-aws -p 8111:8111 -p 4568:4568 -p 4572:4572 -p 4582:4582 -p 4571:4571 -p 4573:4573 -p 6000:6000
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=vector-test-integration-aws --name vector_ec2_metadata \
+	 timberiodev/mock-ec2-metadata:latest
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=vector-test-integration-aws --name vector_localstack_aws \
+	 -e SERVICES=kinesis:4568,s3:4572,cloudwatch:4582,elasticsearch:4571,firehose:4573 \
+	 localstack/localstack@sha256:f21f1fc770ee4bfd5012afdc902154c56b7fb18c14cf672de151b65569c8251e
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=vector-test-integration-aws --name vector_mockwatchlogs \
+	 -e RUST_LOG=trace luciofranco/mockwatchlogs:latest
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create vector-test-integration-aws
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=vector-test-integration-aws -p 8111:8111 --name vector_ec2_metadata \
+	 timberiodev/mock-ec2-metadata:latest
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=vector-test-integration-aws -p 4568:4568 -p 4572:4572 \
+	 -p 4582:4582 -p 4571:4571 -p 4573:4573 --name vector_localstack_aws \
+	 -e SERVICES=kinesis:4568,s3:4572,cloudwatch:4582,elasticsearch:4571,firehose:4573 \
+	 localstack/localstack@sha256:f21f1fc770ee4bfd5012afdc902154c56b7fb18c14cf672de151b65569c8251e
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=vector-test-integration-aws -p 6000:6000 --name vector_mockwatchlogs \
+	 -e RUST_LOG=trace luciofranco/mockwatchlogs:latest
+endif
+
+stop-integration-aws:
+	$(CONTAINER_TOOL) rm --force vector_ec2_metadata vector_mockwatchlogs vector_localstack_aws 2>/dev/null; true
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) stop --name=vector-test-integration-aws 2>/dev/null; true
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm --force --name=vector-test-integration-aws 2>/dev/null; true
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm vector-test-integration-aws 2>/dev/null; true
+endif
+
 test-integration-aws: ## Runs AWS integration tests
 ifeq ($(AUTOSPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose up -d dependencies-aws
-	sleep 5 # Many services are very lazy... Give them a sec...
+	$(MAKE) -k stop-integration-aws \
+    ; rc=$$? \
```

Comment:
What does this do? `rc=$$?` part

Reply:
The stop allows us to ensure the infrastructure is down before we initiate the tests. If the stop fails then make fails - the rc captures that failure and ignores it.

Reply:
We can prefix the command with `-` to get the same effect:

```make
- $(MAKE) -k stop-integration-aws
```

Reply:
Ah thanks. Neat.

Reply:
Whoaaaaa